ui/admin/app/components/form/field/list-wrapper/key-value/index.hbs

```diff
@@ -2,40 +2,78 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: MPL-2.0
 }}
-{{#let (unique-id) (unique-id) as |keyHeaderID valueHeaderID|}}
 
+{{#let (unique-id) (unique-id) as |keyHeaderID valueHeaderID|}}
   <Hds::Table class='list-wrapper-field' name={{@name}}>
     <:head as |H|>
       <H.Tr>
-        <H.Th id={{keyHeaderID}}>{{t 'form.key.label'}}</H.Th>
-        <H.Th id={{valueHeaderID}}>{{t 'form.value.label'}}</H.Th>
+        <H.Th id={{keyHeaderID}}>
+          {{#if @keyLabel}}
+            {{@keyLabel}}
+          {{else}}
+            {{t 'form.key.label'}}
+          {{/if}}</H.Th>
+        <H.Th id={{valueHeaderID}}>
+          {{#if @valueLabel}}
+            {{@valueLabel}}
+          {{else}}
+            {{t 'form.value.label'}}
+          {{/if}}
+        </H.Th>
         <H.Th>
           {{t 'titles.actions'}}
-
         </H.Th>
       </H.Tr>
     </:head>
     <:body as |B|>
       {{#each this.options as |option index|}}
         <B.Tr>
           <B.Td>
-            <Hds::Form::TextInput::Field
-              @value={{option.key}}
-              @type='text'
-              disabled={{@disabled}}
-              aria-labelledby={{keyHeaderID}}
-              {{on 'input' (set-from-event option 'key')}}
-            />
+            {{yield
+              (hash
+                text=(component
+                  'form/field/list-wrapper/key-value/text'
+                  disabled=@disabled
+                  value=option.key
+                  ariaLabel=keyHeaderID
+                  setContext=(set-from-event option 'key')
+                )
+                select=(component
+                  'form/field/list-wrapper/key-value/select'
+                  disabled=@disabled
+                  value=option.key
+                  selectOptions=@selectOptions
```

Comment:
Should these `selectOptions` still be here? Shouldn't they be passed in directly now instead of from `KeyValue`?